File: app/src/main/java/com/kota/hh/ui/main/MainActivity.java
package com.kota.hh.ui.main;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import com.kota.hh.R;
import com.kota.hh.ui.base.BaseActivity;

import junit.framework.Test;

import java.util.List;

import javax.inject.Inject;

public class MainActivity extends BaseActivity implements MainMvpView {

    private static final String EXTRA_TRIGGER_SYNC_FLAG =
            "uk.co.ribot.androidboilerplate.ui.main.MainActivity.EXTRA_TRIGGER_SYNC_FLAG";

    @Inject MainPresenter mMainPresenter;

    public static Intent getStartIntent(Context context, boolean triggerDataSyncOnCreate) {
        Intent intent = new Intent(context, MainActivity.class);
        intent.putExtra(EXTRA_TRIGGER_SYNC_FLAG, triggerDataSyncOnCreate);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        activityComponent().inject(this);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        mMainPresenter.detachView();
    }

    /***** MVP View methods implementation *****/

    @Override
    public void showRibots(List<Test> ribots) {
    }

    @Override
    public void showError() {
    }

    @Override
    public void showRibotsEmpty() {
    }

}

File: app/src/main/java/com/kota/hh/ui/main/MainActivity.kt
package com.kota.hh.ui.main

import android.os.Bundle
import android.view.View
import com.kota.hh.R
import com.kota.hh.ui.base.BaseActivity
import com.kota.hh.ui.main.auth.AuthActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity: BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        showAuthButton.setOnClickListener(onShowAuthButtonClick)
    }

    private val onShowAuthButtonClick = View.OnClickListener {
        startActivity(AuthActivity.createIntent(applicationContext))
    }

}